sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -115,6 +115,12 @@ export class ApplicationController extends BaseController {
       throw new NotFoundException("Student Application not found.");
     }
 
+    if (ApplicationStatus.draft !== applicationToSubmit.applicationStatus) {
```

Comment:
The method below "submitApplication" is already ensuring that the application will be in the draft state and throwing an exception.
I would recommend:
1 - From the service layer, change the current exception being thown 
from
```
if (!application) {
  throw new Error(
    "Student Application not found or it is not in the correct status to be submitted.",
  );
}
```

to

```
throw new CustomNamedError(
    "Student Application not found or it is not in the correct status to be submitted.",
    APPLICATION_NOT_FOUND,
  );
```

2 - Capture the exception in the controller and throw a NotFoundException.

Reply:
The error handling part is done as discussed.